include/ccf/service/tables/jwt.h

```diff
@@ -37,34 +37,51 @@ namespace ccf
   using JwtIssuer = std::string;
   using JwtKeyId = std::string;
   using Cert = std::vector<uint8_t>;
+  using PublicKey = std::vector<uint8_t>;
 
   struct OpenIDJWKMetadata
   {
-    Cert cert;
+    PublicKey public_key;
     JwtIssuer issuer;
     std::optional<JwtIssuer> constraint;
   };
   DECLARE_JSON_TYPE_WITH_OPTIONAL_FIELDS(OpenIDJWKMetadata);
-  DECLARE_JSON_REQUIRED_FIELDS(OpenIDJWKMetadata, cert, issuer);
+  DECLARE_JSON_REQUIRED_FIELDS(OpenIDJWKMetadata, issuer, public_key);
   DECLARE_JSON_OPTIONAL_FIELDS(OpenIDJWKMetadata, constraint);
 
-  using JwtIssuers = ServiceMap<JwtIssuer, JwtIssuerMetadata>;
-  using JwtPublicSigningKeys =
+  using JwtPublicSigningKeysMetadata =
     ServiceMap<JwtKeyId, std::vector<OpenIDJWKMetadata>>;
 
+  struct OpenIDJWKMetadataLegacy
+  {
+    Cert cert;
+    JwtIssuer issuer;
+    std::optional<JwtIssuer> constraint;
+  };
+  DECLARE_JSON_TYPE_WITH_OPTIONAL_FIELDS(OpenIDJWKMetadataLegacy);
+  DECLARE_JSON_REQUIRED_FIELDS(OpenIDJWKMetadataLegacy, issuer, cert);
+  DECLARE_JSON_OPTIONAL_FIELDS(OpenIDJWKMetadataLegacy, constraint);
+
+  using JwtPublicSigningKeysMetadataLegacy =
+    ServiceMap<JwtKeyId, std::vector<OpenIDJWKMetadataLegacy>>;
+
+  using JwtIssuers = ServiceMap<JwtIssuer, JwtIssuerMetadata>;
+
   namespace Tables
   {
     static constexpr auto JWT_ISSUERS = "public:ccf.gov.jwt.issuers";
 
     static constexpr auto JWT_PUBLIC_SIGNING_KEYS_METADATA =
-      "public:ccf.gov.jwt.public_signing_keys_metadata";
+      "public:ccf.gov.jwt.public_signing_keys_metadata_v2";
```

Comment:
nit: I notice `legacy`, `v2` and `extended` being appended to the new/old fields to refer to the new/old versions of the fields, maybe having a convention and only using `legacy` on the new fields may be a bit more clear but feel free to ignore.

Reply:
Unfortunately I can't rename the old table, because there are entries in it in older code version. You can look up a CHANGELOG for this change, as well as [for 5809](https://github.com/microsoft/CCF/blob/main/CHANGELOG.md#authentication), but long store short, we use that older values during code upgrade if the new table hasn't been yet populated, but the request comes to node with the new code.